src/main/java/com/autotune/common/utils/ExponentialBackOff.java

```diff
@@ -0,0 +1,190 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, IBM Corporation and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.common.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * This builder class helps in retrying specific task using exponential backoff approach.
+ */
+public final class ExponentialBackOff {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExponentialBackOff.class);
+    // Maximum total time in Millis to complete exponential backoff approach. Default set to 10sec
+    public static long defaultMaxElapsedTimeMillis = 10 * 1000;   //1o Seconds
+    // Default time to wait for each retry
+    public static long defaultTimeToWait = 1000;
+    // increases the back off period for each retry attempt using a randomization function that grows exponentially.
+    // randomized_interval = retryIntervalMillis * (random value in range [1 - randomization_factor, 1 + randomization_factor])
+    public static double defaultRandomizationFactor = 0.5;
+    // Some task need initial sleep before proceedings . defaults to zero.
+    public static long defaultInitialIntervalMillis = 0;
+    //  increases the back off period for each retry attempt using a randomization function that grows exponentially.
+    public static double defaultMultiplier = 0.5;
+    // Set retry after backoff exhausted with MaxElapsedTimeMillis , default is set to zero that means no retries.
+    private final int numberOfRetries;
+    // Some task need initial sleep before proceedings .
+    private final long initialIntervalMillis;
+    // Maximum total time in Millis to complete exponential backoff approach.
+    private final long maxElapsedTimeMillis;
+    private final Random random = new Random();
+    //  increases the back off period for each retry attempt using a randomization function that grows exponentially.
+    private final double randomizationFactor;
+    private final double multiplier;
+    private int numberOfTriesLeft;
+    private long retryIntervalMillis;
+    private long randomizedIntervalMillis;
+    private double totalRetryIntervalMillis = 0;
+
+
+    public ExponentialBackOff(Builder builder) {
```

Comment:
`null` check for `builder` can avoid Null pointer exception when it access the attributes of builder object. Also we can create a new builder object if null value is passed.